examples/src/main/java/com/datastax/oss/driver/examples/mapper/KillrVideoMapperExample.java

```diff
@@ -0,0 +1,148 @@
+import com.datastax.oss.driver.examples.mapper.killrvideo.video.VideoDao;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Uses the driver's object mapper to interact with a schema.
+ *
+ * <p>We use the data model of the <a href="https://killrvideo.github.io/">KillrVideo</a> sample
+ * application. The mapped entities and DAOs are in the {@link
+ * com.datastax.oss.driver.examples.mapper.killrvideo} package. We only cover a subset of the data
+ * model (ratings, stats, recommendations and comments are not covered).
+ *
+ * <p>Preconditions:
+ *
+ * <ul>
+ *   <li>An Apache Cassandra(R) cluster is running and accessible through the contacts points
+ *       identified by basic.contact-points (see application.conf).
+ *   <li>that cluster has a {@code killrvideo} keyspace with the KillrVideo tables (see {@code
+ *       src/main/resources/killrvideo_schema.cql} in this project).
+ *   <li>the tables are empty.
+ * </ul>
+ *
+ * <p>Side effects:
+ *
+ * <ul>
+ *   <li>inserts data in the tables.
+ * </ul>
+ *
+ * @see <a href="https://docs.datastax.com/en/developer/java-driver/latest/manual/mapper/">Java
+ *     Driver Mapper manual</a>
+ */
+public class KillrVideoMapperExample {
+
+  private static final CqlIdentifier KEYSPACE_ID = CqlIdentifier.fromCql("killrvideo");
+
+  public static void main(String[] args) {
+    try (CqlSession session = CqlSession.builder().build()) {
+      KillrVideoMapper mapper =
+          KillrVideoMapper.builder(session).withDefaultKeyspace(KEYSPACE_ID).build();
+
+      // Create a new user
+      UserDao userDao = mapper.userDao();
+
+      Optional<User> maybeUser =
+          userDao.create("test", "user", "testuser@example.com", "password123".toCharArray());
+      if (!maybeUser.isPresent()) {
+        throw new AssertionError("User creation should succeed (expected empty tables)");
+      }
+      User user = maybeUser.get();
+      System.out.println("Created " + user);
+
+      // Creating another user with the same email should fail
+      assert !userDao
+          .create("test2", "user", "testuser@example.com", "secret123".toCharArray())
+          .isPresent();
+
+      // Simulate login attempts
+      tryLogin(userDao, "testuser@example.com", "password123");
+      tryLogin(userDao, "testuser@example.com", "secret123");
+
+      // Insert a video
+      VideoDao videoDao = mapper.videoDao();
+
+      Video video = new Video();
+      video.setUserid(user.getUserid());
+      video.setName(
+          "Getting Started with DataStax Apache Cassandra as a Service on DataStax Constellation");
+      video.setLocation("https://www.youtube.com/watch?v=68xzKpcZURA");
+      Set<String> tags = new HashSet<>();
+      tags.add("apachecassandra");
+      tags.add("nosql");
+      tags.add("hybridcloud");
+      video.setTags(tags);
+
+      videoDao.create(video);
```

Comment:
Why the decision about having inconsistent API between `videoDao.create` and `UserDao.create`?
The first method is taking the whole Entity object whereas the second one is not taking the user as the argument but single fields of the User entity.

Reply:
Honestly I didn't give it much thought, I'll see if I can improve it.